app/src/main/java/org/listenbrainz/android/util/Utils.kt

```diff
@@ -35,6 +34,11 @@ object Utils {
         return  "https://archive.org/download/mbid-${caaReleaseMbid}/mbid-${caaReleaseMbid}-${caaId}_thumb${size}.jpg"
     }
     
+    fun <T> logAndReturn(it: Throwable) : Resource<T> {
+        e(message = "SocialRepository: ${it.message.toString()}")
```

Comment:
Please keep the `SocialRepository` part dynamic for better usage.

Reply:
Done.